src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php

```diff
@@ -630,15 +630,19 @@ public function withAggregate($relations, $column, $function = null)
             $relation = $this->getRelationWithoutConstraints($name);
 
             if ($function) {
-                $hashedColumn = $this->getRelationHashedColumn($column, $relation);
+                if ($this->getGrammar()->isExpression($column)) {
```

Comment:
Hey @tpetry please correct me if I'm wrong, but I believe this should be `$this->getQuery()->getGrammar()`, instead of `$this->getGrammar()` directly, right?